actions/registerRealm.ts

```diff
@@ -0,0 +1,44 @@
+import { PublicKey, Transaction, TransactionInstruction } from '@solana/web3.js'
+import BN from 'bn.js'
+import { MintMaxVoteWeightSource } from '../models/accounts'
+import { withCreateRealm } from '../models/withCreateRealm'
+import { RpcContext } from '../models/core/api'
+import { sendTransaction } from '../utils/send'
+
+export async function registerRealm(
+  { connection, wallet, programId, programVersion, walletPubkey }: RpcContext,
+  name: string,
+  communityMint: PublicKey,
+  councilMint: PublicKey | undefined,
+  communityMintMaxVoteWeightSource: MintMaxVoteWeightSource,
+  minCommunityTokensToCreateGovernance: BN
+): Promise<string> {
+  const instructions: TransactionInstruction[] = []
+
+  await withCreateRealm(
+    instructions,
+    programId,
+    programVersion,
+    name,
+    walletPubkey,
+    communityMint,
+    walletPubkey,
+    councilMint,
+    communityMintMaxVoteWeightSource,
+    minCommunityTokensToCreateGovernance,
+    undefined
+  )
+
+  const transaction = new Transaction()
+  transaction.add(...instructions)
+
+  return sendTransaction({ transaction, wallet, connection })
+}
+
+export async function getRealmIdFromTransaction(
```

Comment:
There is no need for this function. `registerRealm` can return the new realm address the same way as `createProposal` does

Reply:
right - i thought it was a bit weird to not return tx-id for clicking on the transaction. it could return both. ill change that but ultimately it will just redirect back to the main page and reload the full list